src/runtime/Types/ClassDerived.cs

```diff
@@ -678,6 +710,127 @@ private static ModuleBuilder GetModuleBuilder(string assemblyName, string module
 
             return moduleBuilder;
         }
+
+        /// <summary>
+        /// Builds a C# attribute based on a python tuple definition (Attibute Type, *attributeArgs, ** attributeKwArgs)
+        /// </summary>
+        static CustomAttributeBuilder GetAttributeBuilder(PyObject attr)
+        {
+            // this is a bit complicated because we want to support both unnamed and named arguments
+            // in C# there is a discrepancy between property setters and named argument wrt attributes
+            // in python there is no difference. But luckily there is rarely a conflict, so we can treat
+            // named arguments and named properties or fields the same way.
+            var tp = new PyTuple(attr);
+            Type attribute = tp[0].As<Type>();
+            if (typeof(Attribute).IsAssignableFrom(attribute) == false)
+            {
+                throw new Exception($"This type cannot be used as an attribute type: {attribute}");
+            }
+
+            var args = tp[1].As<PyObject[]>();
+            var dict = new PyDict(tp[2]);
+            var dict2 = new Dictionary<string, PyObject>();
+            foreach (var key in dict.Keys())
+            {
+                var k = key.As<string>();
+                dict2[k] = dict[key];
+            }
+
+            var allConstructors = attribute.GetConstructors();
+            foreach (var constructor in allConstructors)
```

Comment:
This reimplements our overloading logic. We definitely do not want to do that. Instead the Python side should use some simple mechanism to set the fields. For example, there could be a `.with(attributes: dict[str, Any]) -> Self` method, that would set values of the fields/properties from the dictionary.

E.g. `DllImport("kernel32").with({"CharSet": Auto})` (just an example, a different way to do this could also be acceptable).

Reply:
I think that might make sense for properties. I just wanted the call to be analogous to how it is invoked in C#, since people will always be finding the C# documentation/ code samples for how those attributes are used. e.g
`
[AttributeTest("UnnamedArg", NamedArg : "x", NamedProperty="y")]
`
This would now be used like this:
```py
AttributeTest("UnnamedArg", NamedArg = "x", NamedProperty="y")
```

But you are suggesting this instead?

```py
AttributeTest("UnnamedArg", NamedArg = "x").with({"NamedProperty": "y"})
```

Reply:
@rmadsen-ks correct. The reason is that we don't want to maintain yet another overload resolution handler. That's one of the most complicated parts of C# that we try to mirror.

Reply:
Ok, so you are not talking so much about changing these specific lines, but rather changing how the data for that is stored?

I am not sure we can use the other overload resolution handler, because that is for runtime overload resolution, right? this is compile/load time and the data we can put into those attributes is limited.

Reply:
@rmadsen-ks since this is running in Python, it is runtime.

I believe you should be able to use `MethodBinder.Bind` which returns a `Binding`.

Reply:
Unless I'm completely confused, The selected overload gets compiled to IL code by System.Reflection.Emit. That is what the returned CustomAttributeBuilder is for.

Reply:
@rmadsen-ks not sure I understand this comment.

`MethodBinder.Bind` performs basically what your loop performs: selects the method from the list of overloads, and returns it + the arguments for it as .NET would see them.